src/main/java/io/jenkins/plugins/checks/github/GitHubChecksContext.java

```diff
@@ -108,4 +111,37 @@ protected boolean hasValidCredentials(final FilteredLog logger) {
     private Optional<GitHubAppCredentials> findGitHubAppCredentials(final String credentialsId) {
         return getScmFacade().findGitHubAppCredentials(getJob(), credentialsId);
     }
+
+    Optional<Long> getId(final String name) {
+        return getAction(name).map(GitHubChecksAction::getId);
+    }
+
+    // New API method
+    ChecksConclusion getConclusion(final String name) {
+        return getAction(name).map(GitHubChecksAction::getConclusion).orElse(ChecksConclusion.NONE);
+    }
+
+    // Implementation using invisible actions
+    private Optional<GitHubChecksAction> getAction(final String name) {
+        return job.getActions(GitHubChecksAction.class)
+                .stream()
+                .filter(a -> a.getName().equals(name))
+                .findFirst();
+    }
+
+    // Implementation fetching info on request
+    private Optional<GitHubChecksAction> getAction(final GitHub gitHub, final String name) throws IOException {
+        return gitHub.getRepository(getRepository())
+                .getCheckRuns(getHeadSha())
+                .toList()
+                .stream()
+                .filter(r -> r.getName() == name)
```

Comment:
I have no idea if this would work - if you send multiple checks with the same name to github does it discard the old ones, or just only show the most recent with this name?